File: papi/lista2/exercicio8.py
print("Aqui calcularemos o ganho mensal baseado no valor hora e nas horas trabalhadas.")

while True:
    try:
        horas_trabalhadas = float(input("Por favor favor informe o número de horas trabalhadas em um mês - use o ponto como separador decimal: "))
        
        if horas_trabalhadas <= 0:
            raise ValueError

        break
    except ValueError:
        print("Por favor informe um número válido e maior que zero.")


while True:
    try:
        valor_hora = float(input("Por favor favor informe o valor hora - use o ponto como separador decimal: "))
        
        if valor_hora <= 0:
            raise ValueError

        break
    except ValueError:
        print("Por favor informe um número válido e maior que zero.")

print("O ganho mensal é:", valor_hora * horas_trabalhadas)

File: papi/lista2/exercicio8.kt
package lista2

import java.math.RoundingMode
import java.text.NumberFormat
import java.util.Locale
import java.text.DecimalFormat

fun main() {
    var valorHora: Double = 0.0
    var horasTrabalhadasMes: Double = 0.0

    while (true) {
        try {
            print("Digite o seu valor hora em reais, usando o ponto como separador decimal: ")
            valorHora = readLine()!!.toDouble()
            if (valorHora <= 0) {
                throw Exception()
            }
            break
        } catch (e: Exception) {
            println("Por favor informe um número válido maior que zero")
        }
    }
    
    while(true) {
        try {
            print("Digite o número de horas trabalhadas no mês, usando o ponto como separador decimal: ")
            horasTrabalhadasMes = readLine()!!.toDouble()
            if (horasTrabalhadasMes <= 0) {
                throw Exception()
            }
            break                    
        } catch (e: Exception) {
            println("Por favor informe um número válido maior que zero")
        }   
    }

    var rendimentoMensal: Double = (valorHora * horasTrabalhadasMes).toBigDecimal().setScale(2, RoundingMode.UP).toDouble()
    val formatter:NumberFormat = DecimalFormat("#0.00");
    val myLocale:Locale = Locale("pt", "BR")
    var rendimentoMensalFormatado = NumberFormat.getNumberInstance(myLocale).format(rendimentoMensal)

    // Ajustando o número de dígitos depois da vírgula para sempre ficarmos
    // sempre com 2 números depois da vírgula
    val posicaoVirgula = rendimentoMensalFormatado.indexOf(",",0)
    if (posicaoVirgula == -1) {
        rendimentoMensalFormatado += ",00"
    } else if (posicaoVirgula == rendimentoMensalFormatado.length - 2 ) {
        rendimentoMensalFormatado += "0"
    }

    print("O rendimento no mês foi: R$")
    println(rendimentoMensalFormatado)
}